docs/questionRuntime/index.md

```diff
@@ -0,0 +1,60 @@
+# The Python Environment for Questions
+
+This page is meant for discussion of details related to the Python environment that questions are run in, along with how to work within this environment for question development. Are you creating questions and finding that PrairieLearn does not have the Python libraries you need? Don't panic! There are a few different options you have for getting the necessary libraries installed that will be discussed below.
+
+## General information
+
+All `server.py` files for questions are executed in a Docker container created from the `prairielearn/prairielearn` image. This image includes the Python version that is bundled with the [latest version of Miniconda](https://docs.conda.io/en/latest/miniconda.html), as well as the packages in [`images/plbase/python-requirements.txt`](https://github.com/PrairieLearn/PrairieLearn/blob/master/images/plbase/python-requirements.txt). The following are basic ways to get additional information about the Python environment.
+
+- To see which packages are installed, run `pip list` in the docker container.
```

Comment:
On second thought, I don't like this, as this includes a lot of packages that we don't specifically install. I want to have it publicly documented that the *only* packages that we guarantee will exist are those in `python-requirements.txt`.

For reference, here's the current output of `pip list` in the container:

```
alabaster                     0.7.13
ansi2html                     1.8.0
anyio                         3.6.2
argon2-cffi                   21.3.0
argon2-cffi-bindings          21.2.0
asttokens                     2.2.1
attrs                         22.2.0
Babel                         2.12.1
backcall                      0.2.0
backports.functools-lru-cache 1.6.4
beautifulsoup4                4.11.2
black                         23.1.0
bleach                        6.0.0
brotlipy                      0.7.0
certifi                       2022.12.7
cffi                          1.15.1
charset-normalizer            2.1.1
chevron                       0.14.0
click                         8.1.3
colorama                      0.4.6
comm                          0.1.2
commonmark                    0.9.1
conda                         23.1.0
conda-package-handling        2.0.2
conda_package_streaming       0.7.0
contourpy                     1.0.7
cryptography                  39.0.1
cycler                        0.11.0
debugpy                       1.6.6
decorator                     5.1.1
defusedxml                    0.7.1
docutils                      0.19
entrypoints                   0.4
et-xmlfile                    1.1.0
exceptiongroup                1.1.0
executing                     1.2.0
fastjsonschema                2.16.3
flake8                        6.0.0
flake8-black                  0.3.6
flake8-isort                  6.0.0
Flake8-pyproject              1.2.2
flit_core                     3.8.0
fonttools                     4.39.0
html2text                     2020.1.16
idna                          3.4
imagesize                     1.4.1
importlib-metadata            6.0.0
importlib-resources           5.12.0
iniconfig                     2.0.0
ipykernel                     6.21.2
ipython                       8.11.0
ipython-genutils              0.2.0
isort                         5.12.0
jedi                          0.18.2
Jinja2                        3.1.2
joblib                        1.2.0
jsonschema                    4.17.3
jupyter_client                8.0.3
jupyter_core                  5.2.0
jupyter-events                0.6.3
jupyter_server                2.3.0
jupyter_server_terminals      0.4.4
jupyterlab-pygments           0.2.2
kiwisolver                    1.4.4
lxml                          4.9.2
MarkupSafe                    2.1.2
matplotlib                    3.7.0
matplotlib-inline             0.1.6
mccabe                        0.7.0
mistune                       2.0.5
mpmath                        1.2.1
mypy-extensions               1.0.0
nbclassic                     0.5.2
nbclient                      0.7.2
nbconvert                     7.2.9
nbformat                      5.7.3
nest-asyncio                  1.5.6
networkx                      3.0
nltk                          3.8.1
notebook                      6.5.3
notebook_shim                 0.2.2
numpy                         1.24.2
openpyxl                      3.1.1
packaging                     23.0
pandas                        1.5.3
pandas-stubs                  1.5.3.230227
pandocfilters                 1.5.0
parso                         0.8.3
pathspec                      0.11.0
patsy                         0.5.3
pexpect                       4.8.0
pickleshare                   0.7.5
Pillow                        9.4.0
Pint                          0.20.1
pip                           23.0
pkgutil_resolve_name          1.3.10
platformdirs                  3.1.0
pluggy                        1.0.0
prometheus-client             0.16.0
prompt-toolkit                3.0.38
psutil                        5.9.4
ptyprocess                    0.7.0
PuLP                          2.7.0
pure-eval                     0.2.2
pycodestyle                   2.10.0
pycosat                       0.6.4
pycparser                     2.21
pycryptodome                  3.17
pydash                        6.0.2
pyflakes                      3.0.1
Pygments                      2.14.0
pygments-ansi-color           0.2.0
pygraphviz                    1.6
pyOpenSSL                     23.0.0
pyparsing                     3.0.9
pyquaternion                  0.9.9
pyrsistent                    0.19.3
PySocks                       1.7.1
pytest                        7.2.1
pytest-lazy-fixture           0.6.3
python-dateutil               2.8.2
python-json-logger            2.0.7
pytz                          2022.7.1
pytz-deprecation-shim         0.1.0.post0
PyYAML                        6.0
pyzmq                         25.0.0
recommonmark                  0.7.1
regex                         2022.10.31
requests                      2.28.2
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
rpy2                          3.5.9
ruamel.yaml                   0.17.21
ruamel.yaml.clib              0.2.7
scikit-learn                  1.2.1
scipy                         1.10.1
Send2Trash                    1.8.0
setuptools                    65.6.3
six                           1.16.0
sniffio                       1.3.0
snowballstemmer               2.2.0
soupsieve                     2.3.2.post1
Sphinx                        6.1.3
sphinx-markdown-builder       0.5.5
sphinxcontrib-applehelp       1.0.4
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.1
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
stack-data                    0.6.2
statsmodels                   0.13.5
sympy                         1.11.1
terminado                     0.17.1
text-unidecode                1.3
threadpoolctl                 3.1.0
tinycss2                      1.2.1
tomli                         2.0.1
toolz                         0.12.0
tornado                       6.2
tqdm                          4.64.1
traitlets                     5.9.0
types-docutils                0.19.1.6
types-Pygments                2.14.0.5
types-pytest-lazy-fixture     0.6.3.3
types-pytz                    2022.7.1.2
types-setuptools              67.5.0.0
typing_extensions             4.5.0
tzdata                        2022.7
tzlocal                       4.2
unify                         0.5
untokenize                    0.1.1
urllib3                       1.26.14
wcwidth                       0.2.6
webencodings                  0.5.1
websocket-client              1.5.1
wheel                         0.38.4
yapf                          0.32.0
zipp                          3.15.0
zstandard                     0.19.0
```

Reply:
Ah that makes sense. Since we already link to the requirements file, I can just delete this line.

Reply:
Removed here https://github.com/PrairieLearn/PrairieLearn/pull/3892/commits/32650f04448a9aa575385dba10889f983dd1f81a